pkg/otelutil/log/logr.go

```diff
@@ -0,0 +1,273 @@
+package log
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/global"
+	semconv "go.opentelemetry.io/otel/semconv/v1.27.0"
+)
+
+// Most of this is heavily modified from:
+// - https://github.com/open-telemetry/opentelemetry-go-contrib/blob/a91d74714434fcf5b54c577036df09e9e1b660e4/bridges/otellogr/logsink.go
+// - https://github.com/open-telemetry/opentelemetry-go-contrib/blob/a91d74714434fcf5b54c577036df09e9e1b660e4/bridges/otellogr/convert.go
```

Comment:
If this modification of logsink and convert what was not working in the otel implementation?

Reply:
The otel bridge implementations:

1. don't set severity levels properly
2. don't handle `Error(nil, ...)` calls properly (it panics)
3. don't allow for setting timestamps
4. don't properly marshal things that implement `String() string` such as UUIDs which are set every reconciliation loop in the logging context.

Reply:
Mind adding that list to the comment for future reference?